tap_s3_csv/__init__.py

```diff
@@ -48,19 +71,24 @@ def do_sync(config, catalog, state):
 
     LOGGER.info('Done syncing.')
 
+
 @singer.utils.handle_top_exception(LOGGER)
-def main():
+def main() -> None:
+    """
+    Main function
+    :return: None
+    """
     args = singer.utils.parse_args(REQUIRED_CONFIG_KEYS)
     config = args.config
 
     # Reassign the config tables to the validated object
     config['tables'] = CONFIG_CONTRACT(config.get('tables', {}))
 
     try:
-        for page in s3.list_files_in_bucket(config['bucket']):
+        for _ in s3.list_files_in_bucket(config['bucket']):
             break
         LOGGER.warning("I have direct access to the bucket without assuming the configured role.")
-    except:
+    except Exception:
```

Comment:
Also, it's a good practice to catch exceptions as specific as possible (especially if you know what exceptions might be there). Here is some good link about it https://wiki.python.org/moin/HandlingExceptions

Reply:
Yes, I know :+1: , the thing though is the code block inside of the  try except statement uses boto3 lib and I didn't have time nor energy on Friday to read all the code and figure out what exception are thrown.